src/Curl/Curl.php

```diff
@@ -357,13 +389,24 @@ public function exec($ch = null)
         if ($this->getOpt(CURLINFO_HEADER_OUT) === true) {
             $this->requestHeaders = $this->parseRequestHeaders(curl_getinfo($this->curl, CURLINFO_HEADER_OUT));
         }
-        $this->responseHeaders = $this->parseResponseHeaders($this->rawResponseHeaders);
-        list($this->response, $this->rawResponse) = $this->parseResponse($this->responseHeaders, $this->rawResponse);
+        if ($this->responseHeadersParsing) {
```

Comment:
Since your adding to all of this, could we not make these protected methods additionally?